pkg/gossiper/db.go

```diff
@@ -19,31 +20,31 @@ func Newdb(path string) (int, *mydb) {
 	newdb := &mydb{}
 	newdb.path = path
 	var err error
-	newdb.db, err = leveldb.OpenFile("path/to/db", nil)
+	newdb.db, err = leveldb.OpenFile(newdb.path, nil)
+	newdb.mybatch = NewBatch()
```

Comment:
Could have merged all these fields for `newdb` Into the first line. Like newdb. := &mudbank{
Mybatch: NewBatch(),
...
}